src/js/jsx/shared/Datalist.jsx

```diff
@@ -393,33 +389,58 @@ define(function (require, exports, module) {
         },
 
         /**
+         * Updates position of autofill using hidden HTML elements.
+         *
+         * @private
+         * @param {string} value The current value of the input
+         * @param {string} icon The SVG icon for the input
+         */
+        _updateAutofillPosition: function (value, icon) {
+            // Base position off of hidden element widths because Datalist sub-components
+            // haven't rendered yet using the new value, and we don't want the suggestion to move
+            // after rendering
+
+            var hiddenInput = React.findDOMNode(this.refs.hiddenTextInput);
```

Comment:
Wow so scary. For some reason I originally thought that moving the autofill suggestion in `componentDidUpdate` would be bad but it looks like that's possible, so I can avoid using `innerHTML` :grin: